README.md

```diff
@@ -44,9 +44,9 @@ import { Router } from '@angular/router';
 })
 export class AppComponent {
     constructor(
-        private router: Router
+        private router: Router,
+        private angularPlugin: AngularPlugin
     ){
-        var angularPlugin = new AngularPlugin();
         const appInsights = new ApplicationInsights({ config: {
```

Comment:
Actually, thinking about this a bit more deeply there is a couple of issues with both the original and "injecting" the angularPlugin instance.

The first issue with the existing example is: -
- While the documentation states that `Set up an instance of Application Insights in the entry component in your app`, *if* you do just this then all is well, however, if people miss this comment it would cause multiple instances of the SDK to be instantiated -- which WILL cause duplication for any of the automatic events (XHR/fetch, Page View (based on URL change, etc)

The second issue with having the Plugin as Injectable
- A Plugin can actually only be initialized once, so it's original "configuration" is based on the config passed to the initial `initialize()` call. While internally, the `BaseTelemetryPlugin` and the `Core` does and can handle using a "pre-initialized" plugin into it's new change (ie. sharing the same plugin with different instances), problems will arise when the original "initializing" core instance "unloads" as this will also cause all of the plugins that it initialized to be "unloaded" also (clearing a lot of the internal values to null / undefined) and when this plugin is used in another instance of the Core it WILL throw (ie. the code doesn't "re-initialize" components if they happen to have been "unloaded").
- There are only 2 times that a plugin is checked for and initialized if it's not already
  - During the `core.initialize()`
  - During the `core.addPlugin()`
- And the `unload` is also triggered via `core.getPlugin().remove()` (if the current `core` instance is the same instance that originally initialized the plugin).

What does all this mean?
Simplistically, if we want / need to support injecting the instance into the `AppComponent` then we should inject the `appInsights` instance and not the plugin. This would then also support having multiple `AppComponent` instances on a page with the same `appInsights` instance.

Reply:
> This would then also support having multiple AppComponent instances on a page

It's not possible to have multiple instances of the same `AppComponent`. This would only be possible if you're running multiple separate apps on the same page that point to different selectors, e.g., `app-root`, `another-app-root`, etc.

Reply:
It's probably worth adding this in the readme that (if) you do this then it will cause event duplication.

Reply:
It might be worth adding unit tests from my side for those cases you mentioned in the comment above too.

Reply:
I revisited your comments and reverted `AngularPlugin` creation back to the 'new' expression style. But it now requires an injector when being constructed so we still use features of the Angular DI system except of relying on implicit static properties (which might be set/reset any time).